File: myolink/src/main/java/com/tech7fox/myolink/processor/imu/ImuProcessor.java
/*
 * Android Myo library by darken
 * Matthias Urhahn (matthias.urhahn@rwth-aachen.de)
 * mHealth - Uniklinik RWTH-Aachen.
 */
package com.tech7fox.myolink.processor.imu;

import com.tech7fox.myolink.processor.BaseDataPacket;
import com.tech7fox.myolink.processor.BaseProcessor;
import com.tech7fox.myolink.services.Imu;

/**
 * Processor that converts {@link BaseDataPacket} object into {@link ImuData} objects.
 */
public class ImuProcessor extends BaseProcessor {
    private static final String TAG = "myolink:ImuProcessor";

    public ImuProcessor() {
        super();
        getSubscriptions().add(Imu.IMUDATA.getCharacteristicUUID());
    }

    @Override
    protected void doProcess(BaseDataPacket packet) {
        ImuData imuData = new ImuData(packet);

        for (DataListener listener : getDataListeners()) {
            ImuDataListener imuListener = (ImuDataListener) listener;
            imuListener.onNewImuData(imuData);
        }
    }


    public interface ImuDataListener extends DataListener {
        void onNewImuData(ImuData imuData);
    }

    public void addListener(ImuDataListener listener) {
        super.addDataListener(listener);
    }
}

File: myolink/src/main/kotlin/com/tech7fox/myolink/processor/imu/ImuProcessor.kt
/*
 * Android Myo library by darken
 * Matthias Urhahn (matthias.urhahn@rwth-aachen.de)
 * mHealth - Uniklinik RWTH-Aachen.
 */
package com.tech7fox.myolink.processor.imu

import com.tech7fox.myolink.processor.BaseDataPacket
import com.tech7fox.myolink.processor.BaseProcessor
import com.tech7fox.myolink.services.Imu

/**
 * Processor that converts [BaseDataPacket] object into [ImuData] objects.
 */
class ImuProcessor : BaseProcessor() {

    init {
        subscriptions.add(Imu.IMUDATA.characteristicUUID)
    }

    override fun doProcess(packet: BaseDataPacket) {
        val imuData = ImuData(packet)

        for (listener in getDataListeners()) {
            val imuListener = listener as ImuDataListener
            imuListener.onNewImuData(imuData)
        }
    }

    interface ImuDataListener : DataListener {
        fun onNewImuData(imuData: ImuData)
    }

    fun addListener(listener: ImuDataListener) {
        super.addDataListener(listener)
    }

    companion object {
        private const val TAG = "myolink:ImuProcessor"
    }
}
